packages/react-urql/src/hooks/useQuery.ts

```diff
@@ -53,17 +53,30 @@ export function useQuery<Data = any, Variables = object>(
   const request = useRequest<Data, Variables>(args.query, args.variables);
   const cache = getCacheForClient(client);
 
+  const currDeps: unknown[] = [
+    client,
+    request,
+    args.pause,
+    args.requestPolicy,
+    args.context,
+  ];
+
   const [meta, setMeta] = useState<{
     source: Source<OperationResult<Data, Variables>> | null;
     prevValue: UseQueryState<Data, Variables>;
-    deps: Array<any>;
+    deps: unknown[];
     suspense: boolean;
-  }>({
-    source: null,
+  }>(() => ({
```

Comment:
This isn't strictly speaking necessary, but let's avoid calling `client.executeQuery` on every render 😆